src/libopenrave-core/generictrajectory.cpp

```diff
@@ -441,6 +442,18 @@ class GenericTrajectory : public TrajectoryBase
             WriteBinaryVector(O, this->_vtrajdata);
 
             WriteBinaryString(O, GetDescription());
+
+            // Readable interfaces, added on VERSION_NUMBER=0x0002
+            const uint16_t numReadableInterfaces = GetReadableInterfaces().size();
+            WriteBinaryUInt16(O, numReadableInterfaces);
+            FOREACHC(itReadableInterface, GetReadableInterfaces()) {
+                WriteBinaryString(O, itReadableInterface->first);  // xmlid
+                std::stringstream ss;
```

Comment:
@strmio will be possibly faster if you take `std::stringstream` out of the scope and do `ss.clear(); ss.str("");` in the loop

Reply:
Sure.